atom/renderer/atom_render_view_observer.cc

```diff
@@ -150,16 +145,24 @@ void AtomRenderViewObserver::OnDestruct() {
 void AtomRenderViewObserver::OnBrowserMessage(bool send_to_all,
                                               const base::string16& channel,
                                               const base::ListValue& args) {
-  if (!document_created_)
-    return;
-
   if (!render_view()->GetWebView())
     return;
 
   blink::WebFrame* frame = render_view()->GetWebView()->MainFrame();
   if (!frame || frame->IsWebRemoteFrame())
     return;
 
+  // Don't handle browser messages before document element is created.
+  // When we receive a message from the browser, we try to transfer it
+  // to a web page, and when we do that Blink creates an empty
+  // document element if it hasn't been created yet, and it makes our init
+  // script to run while `window.location` is still "about:blank".
+  blink::WebDocument document = frame->GetDocument();
+  blink::WebElement html_element = document.DocumentElement();
+  if (html_element.IsNull()) {
```

Comment:
Yeah it should be the same, Chromium is in the process of deprecating `RenderViewObserver` for `RenderFrameObserver`, so eventually we should migrate code of `atom_render_view_observer.cc` to use `atom_render_frame_observer.cc`.